toolset/utils/docker_helper.py

```diff
@@ -10,98 +9,104 @@
+                        index = buffer.index("\n")
+                        line = buffer[:index]
+                        buffer = buffer[index + 1:]
+                        log(line,
+                            prefix=log_prefix,
+                            file=build_log,
+                            color=Fore.WHITE + Style.BRIGHT \
+                                if re.match(r'^Step \d+\/\d+', line) else '')
+
+                if buffer:
+                    log(buffer,
                         prefix=log_prefix,
                         file=build_log,
                         color=Fore.WHITE + Style.BRIGHT \
-                            if re.match(r'^Step \d+\/\d+', line) else '')
-
-            if buffer:
-                log(buffer,
+                            if re.match(r'^Step \d+\/\d+', buffer) else '')
+            except Exception:
+                tb = traceback.format_exc()
+                log("Docker build failed; terminating",
                     prefix=log_prefix,
                     file=build_log,
-                    color=Fore.WHITE + Style.BRIGHT \
-                        if re.match(r'^Step \d+\/\d+', buffer) else '')
-        except Exception:
-            tb = traceback.format_exc()
-            log("Docker build failed; terminating",
-                prefix=log_prefix,
-                file=build_log,
-                color=Fore.RED)
-            log(tb, prefix=log_prefix, file=build_log)
-            raise
-
-
-def build(benchmarker_config, test_names, build_log_dir=os.devnull):
-    '''
-    Builds the test docker containers
-    '''
-    tests = gather_tests(
-        include=test_names, benchmarker_config=benchmarker_config)
-
-    for test in tests:
+                    color=Fore.RED)
+                log(tb, prefix=log_prefix, file=build_log)
+                self.benchmarker.timeLogger.log_build_end(
+                    log_prefix=log_prefix,
+                    file=build_log)
+                raise
+
+            self.benchmarker.timeLogger.log_build_end(
+                log_prefix=log_prefix,
+                file=build_log)
+
+    def clean(self):
+        '''
+        Cleans all the docker images from the system
+        '''
+
+        self.server.images.prune()
+        for image in self.server.images.list():
+            if len(image.tags) > 0:
+                # 'techempower/tfb.test.gemini:0.1' -> 'techempower/tfb.test.gemini'
+                image_tag = image.tags[0].split(':')[0]
+                if image_tag != 'techempower/tfb' and 'techempower' in image_tag:
+                    self.server.images.remove(image.id, force=True)
+        self.server.images.prune()
+
+        self.database.images.prune()
+        for image in self.database.images.list():
+            if len(image.tags) > 0:
+                # 'techempower/tfb.test.gemini:0.1' -> 'techempower/tfb.test.gemini'
+                image_tag = image.tags[0].split(':')[0]
+                if image_tag != 'techempower/tfb':
```

Comment:
I think that this still needs to check for the `'techempower' in image_tag` in the case of single-machine setups (otherwise, we may be removing images on the host machine unrelated to the project, and that should be avoided.

Reply:
Agreed. I hadn't yet figured out why we weren't doing it [here](https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/toolset/utils/docker_helper.py#L43) but will update.